src/main/java/frc/robot/subsystems/Climber.java

```diff
@@ -0,0 +1,223 @@
+  private TalonFXConfiguration frontConfigs;
+  private TalonFXConfiguration backConfigs;
+  private DCMotor simClimberGearbox;
+  private SingleJointedArmSim climberSimNormal;
+  private SingleJointedArmSim climberSimLifting;
+  private Mechanism2d climbMech;
+  private MechanismRoot2d climbMechRoot;
+  private MechanismLigament2d climbMechLigament;
+  private MechanismLigament2d climbMechTargetLigament;
+  private Slot0Configs climberConfigsToClimber;
+  private Slot1Configs climberConfigsAtClimber;
+  private Angle climberTargetAngle;
+  private boolean isClimbing;
+
+  public Climber() {
+    climberMotorFront = new TalonFX(MotorIDConstants.kCLimberMotorLeader);
+    climberMotorBack = new TalonFX(MotorIDConstants.kCLimberMotorFollower);
+    climberConfigsToClimber =
+        new Slot0Configs()
+            .withKP(ClimberConstants.kClimberP0)
+            .withKI(ClimberConstants.kClimberI0)
+            .withKD(ClimberConstants.kClimberD0)
+            .withKG(ClimberConstants.kClimberkG0)
+            .withKV(ClimberConstants.kClimberkV0);
+    climberConfigsAtClimber =
+        new Slot1Configs()
+            .withKP(ClimberConstants.kClimberP1)
+            .withKI(ClimberConstants.kClimberI1)
+            .withKD(ClimberConstants.kClimberD1)
+            .withKG(ClimberConstants.kClimberkG1)
+            .withKV(ClimberConstants.kClimberkV1);
+    isClimbing = false;
+    // Set the configs
+    climberOutputConfigs = new MotorOutputConfigs().withNeutralMode(NeutralModeValue.Brake);
+    frontConfigs =
+        new TalonFXConfiguration()
+            .withSlot0(climberConfigsToClimber)
+            .withSlot1(climberConfigsAtClimber)
+            .withMotorOutput(
+                climberOutputConfigs.withInverted(InvertedValue.CounterClockwise_Positive));
+    backConfigs =
+        new TalonFXConfiguration()
+            .withSlot0(climberConfigsToClimber)
+            .withSlot1(climberConfigsAtClimber)
+            .withMotorOutput(climberOutputConfigs.withInverted(InvertedValue.Clockwise_Positive));
+    climberMotorFront.getConfigurator().apply(frontConfigs);
+    climberMotorBack.getConfigurator().apply(backConfigs);
+    // For simulation
+    climberTargetAngle = ClimberConstants.kClimberRetractedSetpoint;
+    if (Robot.isSimulation()) {
+      simClimberGearbox = DCMotor.getKrakenX60(2);
+      climberSimNormal =
+          new SingleJointedArmSim(
+              simClimberGearbox,
+              ClimberConstants.KGearRatio,
+              SingleJointedArmSim.estimateMOI(
+                  ClimberConstants.kClimberArmLength.in(Meters),
+                  ClimberConstants.kClimberMass.in(Kilograms)),
+              ClimberConstants.kClimberArmLength.in(Meters),
+              ClimberConstants.kClimberArmMinAngle.in(Radians),
+              ClimberConstants.kClimberArmMaxAngle.in(Radians),
+              true,
+              climberTargetAngle.in(Radians));
+      climberSimLifting =
```

Comment:
I don't think i fully understand why we need to SingleJointedArmSim's here. I seems like it would be more accurate to the real robot to have only one that you then see move slower/faster after a certain point.

Reply:
I asked him to add two to test the two stages of motion:
1. Before the arms contact the cage
2. Once the arms contact the cage and need to lift the robot's weight.

The physics is significantly different between these stages, and is why there are two PIDF Slots being used. 

Jameson should add some comments in the code about how the simulation is setup to make it easier to understand what is going on.